code/core/src/preview-api/modules/preview-web/render/mount-utils.ts

```
@@ -4,32 +4,70 @@ export function mountDestructured(playFunction?: (...args: any[]) => any): boole
-  const match = fn.toString().match(/[^(]*\(([^)]*)/);
-
-  if (!match) {
+/**
+ * Extracts a list of properties destructured from the argument of a play function, either inline or
+ * as the first statement in the body of the function.
+ *
+ * @param fn - The function to extract the properties from.
+ * @returns An array of property names.
+ */
+export function getUsedProps(fn: (...args: unknown[]) => unknown) {
+  const [, args, body] = fn.toString().match(/[^(]*\(([^)]+)\)(?:.*{([^]+)})?/) || [];
```

Biome (2.1.2)

[error] The regular expression includes this negated empty character class.

Negated empty character classes match anything.
If you want to match against [, escape it \[.
Otherwise, remove the character class or fill it.

(lint/correctness/noEmptyCharacterClassInRegex)